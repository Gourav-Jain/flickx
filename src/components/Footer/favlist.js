import React from 'react';
import { Icon, Popup, Message, List, Image } from 'semantic-ui-react';
import { API_IMAGE_URL } from '../../utils'

export const FavoriteList = (props) => {
    const { items, clicked } = props

    return (
        <>
            {items ? <List divided verticalAlign='middle'>
                {items.map((val, key) => (
                    <List.Item>
                        <List.Content floated='right'>
                            <Popup
                                trigger={<a key={key} href onClick={() => clicked(key)}><Icon name='close' color={items[key].wl ? "blue" : "grey"} /></a>}
                                content="Remove"
                                position='right center'
                            />
                        </List.Content>
                        <Image avatar src={`${API_IMAGE_URL}${items[key].poster_path}`} />
                        <List.Content>
                            <List.Header>{items[key].title}</List.Header>
                            {items[key].release_date}
                        </List.Content>
                    </List.Item>

                ))}
            </List> :
                <Message warning>
                    <p>No videos in your watch later list</p>
                </Message>}
        </>
    )
}
