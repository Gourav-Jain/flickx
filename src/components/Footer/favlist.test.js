import React from "react";
import { shallow } from "enzyme";

import { FavoriteList } from "./favlist";

describe("<FavoriteList />", () => {
    const props = {
        items: [],
        clicked: jest.fn
    };

    beforeEach(jest.clearAllMocks);

    describe("@actions", () => {
        test("should display fav list", () => {
            const favlist = shallow(<FavoriteList {...props} />);
            expect(favlist).toMatchSnapshot();
        });
    });
});