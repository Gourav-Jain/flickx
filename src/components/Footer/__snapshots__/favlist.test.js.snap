// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteList /> @actions should display fav list 1`] = `ShallowWrapper {}`;
